### PR TITLE
Keep track of consent type given

### PR DESCRIPTION
**Pull request by arothuis**
To determine what type of consent a user has given for a certain service, this should be stored. 

This way, it is possible to inform the user about whether he has implicitly or explicitly consented to releasing his data to a specific service.

Part of: Profile [ticket #106401530](https://www.pivotaltracker.com/story/show/106401530)

### CODE REVIEW

**Comment by arothuis** [comment]
@DRvanR, @rjkip

File: library/EngineBlock/Corto/Module/Service/ProvideConsent.php [hunk 1]
```diff
@@ -60,10 +60,17 @@ public function serve($serviceName)
             $log = $application->getLogInstance();
             $log->info('Raw HTTP request', array('http_request' => (string) $application->getHttpRequest()));
         }
+        $serviceProviderMetadata = $spMetadataChain[0];
 
-        if ($this->isConsentDisabled($spMetadataChain, $identityProvider))   {
-            $response->setConsent(SAML2_Const::CONSENT_INAPPLICABLE);
+        $attributes = $response->getAssertion()->getAttributes();
+        $consent = $this->_consentFactory->create($this->_server, $response, $attributes);
```
Comment (DRvanR) [inline library/EngineBlock/Corto/Module/Service/ProvideConsent.php]: `$consentRepositiry`? `$consentService`?

File: library/EngineBlock/Corto/Model/Consent.php [hunk 2]
```diff
@@ -60,12 +71,13 @@ public function hasStoredConsent(ServiceProvider $serviceProvider)
...
 
-            $query = "SELECT * FROM {$this->_tableName} WHERE hashed_user_id = ? AND service_id = ? AND attribute = ?";
+            $query = "SELECT * FROM {$this->_tableName} WHERE hashed_user_id = ? AND service_id = ? AND attribute = ? AND consent_type = ?";
             $hashedUserId = sha1($this->_getConsentUid());
             $parameters = array(
                 $hashedUserId,
                 $serviceProvider->entityId,
-                $attributesHash
+                $attributesHash,
+                $consentType,
```
Comment (rjkip) [inline library/EngineBlock/Corto/Model/Consent.php]: I don't like how `ConsentType::toString()`, `ConsentType::__construct()` (see ConsentRepository) and the `Model_Consent::EXPLICIT` and `IMPLICIT` constants are coupled. I'd rather see the database values decoupled from the value object's constructor and toString() method.

**Comment by rjkip** [comment]
One remark. The rest: :+1::rocket::ship:

**Comment by DRvanR** [comment]
:+1:

**Pull request merged**